testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/mock/JobWorkerMockImpl.java

```diff
@@ -64,23 +68,34 @@ public void thenThrowBpmnError(final String errorCode) {
   public void thenThrowBpmnError(final String errorCode, final Map<String, Object> variables) {
     withHandler(
         (jobClient, job) -> {
+          LOGGER.debug(
+              "Mock: Throw BPMN error with error code {} and variables {} [job-type: '{}', job-key: '{}']",
+              errorCode,
+              variables,
+              jobType,
+              job.getKey());
           jobClient
               .newThrowErrorCommand(job)
               .errorCode(errorCode)
               .variables(variables)
               .send()
               .join();
-          LOGGER.debug(
-              "Mocked job {} with variables {} threw a BPMN error with error code {}",
-              job.getKey(),
-              variables,
-              errorCode);
         });
   }
 
   @Override
   public void withHandler(final JobHandler jobHandler) {
```

Comment:
🔧 Just a small refactoring suggestion to keep the actual invocation free of clutter.

```suggestion
  public void withHandler(final JobHandler jobHandler) {
    final JobHandler loggingJobHandler = (client, job) -> {
      LOGGER.debug(
          "Mock: Pass job to custom handler [job-type: '{}', job-key: '{}']",
          jobType,
          job.getKey());

      jobHandler.handle(client, job);
    };

    client.newWorker().jobType(jobType).handler(loggingJobHandler).open();
  }
```